src/redux/services/projectConfig.ts

```diff
@@ -158,32 +158,8 @@ export const updateProjectSettings = (dispatch: (action: AnyAction) => void, pro
 };
 
 export const mergeConfigs = (baseConfig: ProjectConfig, config?: ProjectConfig | null): ProjectConfig => {
-  if (!(baseConfig && baseConfig.settings && baseConfig.kubeConfig)) {
-    throw Error('Base config must be set');
-  }
-
-  if (!config) {
-    return baseConfig;
-  }
-
-  const serializedBaseConfig: SerializableObject = flatten<any, any>(baseConfig);
-  const serializedConfig: SerializableObject = flatten<any, any>(config);
-
-  // override base properties
-  Object.keys(serializedBaseConfig).forEach((key: string) => {
-    if (!_.isUndefined(serializedConfig[key])) {
-      serializedBaseConfig[key] = serializedConfig[key];
-    }
-  });
-
-  // add missing base properties
-  Object.keys(serializedConfig).forEach((key: string) => {
-    if (_.isUndefined(serializedBaseConfig[key])) {
-      serializedBaseConfig[key] = serializedConfig[key];
-    }
-  });
-
-  return flatten.unflatten(serializedBaseConfig);
+  invariant(baseConfig && baseConfig.settings && baseConfig.kubeConfig, 'base config expected');
+  return mergeWith(baseConfig, config, (_a, b) => (isArray(b) ? b : undefined));
```

Comment:
This should fix crash due to incorrect merging due to parent in the middle not existing in base config.

```
// base.kubeConfig.context is undefined
// conf.kubeConfig.kubeConfig.contexts.0 // is defined
```

`flatten.unflatten` will fail while `mergeWith` will not.